src/WebJobs.Extensions.DurableTask/EntityScheduler/EntityId.cs

```diff
@@ -60,7 +68,7 @@ internal static EntityId GetEntityIdFromSchedulerId(string schedulerId)
         /// <inheritdoc/>
         public override string ToString()
         {
-            return GetSchedulerIdFromEntityId(this);
+            return this.schedulerId;
```

Comment:
I'm not 100% sure because I don't always understand how JSON.NET works, but if this object is deserialized from JSON, is it possible that `schedulerId` will be null (because the constructor doesn't necessarily get called)?  I ask because I see comments in the `EntityName` and `EntityKey` properties saying we still need the private setters for deserialization.

Reply:
Good point!

There are two approaches I can take. One of which is lazily calculate this field when I call ToString() the first time, the other is to serialize the field. I think I prefer the first, as I don't see a reason we should send this field around since it can already be derived.

Reply:
I went with the first approach.